webapp/views/App/views/Data/MapView/CoordinateAttributeDataLayer/CoordinateAttributePolygon.js

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from 'react'
+import { Rectangle, Circle } from 'react-leaflet'
+import GeometryUtil from 'leaflet-geometryutil'
+import L from 'leaflet'
+import { useMapContextOptions } from '@webapp/components/Map/MapContext'
+import { useSurveyInfo } from '@webapp/store/survey'
+import _ from 'lodash'
+import PropTypes from 'prop-types'
+
+export const CoordinateAttributePolygon = (props) => {
+  const surveyInfo = useSurveyInfo()
+  const polygonProperties = JSON.parse(surveyInfo.props.samplingPolygon)
```

Comment:
for already existing surveys, surveyInfo.props.samplingPolygon is null, please check that, or use a getter that returns an empty object in that case. Also the JSON parsing shouldn't be needed because surveyPolygon should be already an object